`OnceCell::try_init` and `OnceLock::initialize` use the `rust-cold` calling convention.

diff -ru a/library/core/src/cell/once.rs b/library/core/src/cell/once.rs
--- a/library/core/src/cell/once.rs
+++ b/library/core/src/cell/once.rs
@@ -313,7 +313,7 @@
     // Avoid inlining the initialization closure into the common path that fetches
     // the already initialized value
     #[cold]
-    fn try_init<F, E>(&self, f: F) -> Result<&T, E>
+    extern "rust-cold" fn try_init<F, E>(&self, f: F) -> Result<&T, E>
     where
         F: FnOnce() -> Result<T, E>,
     {
diff -ru a/library/core/src/lib.rs b/library/core/src/lib.rs
--- a/library/core/src/lib.rs
+++ b/library/core/src/lib.rs
@@ -151,6 +151,7 @@
 #![feature(pin_macro_internals)]
 #![feature(prelude_import)]
 #![feature(repr_simd)]
+#![feature(rust_cold_cc)]
 #![feature(rustc_attrs)]
 #![feature(rustdoc_internals)]
 #![feature(simd_ffi)]
diff -ru a/library/std/src/lib.rs b/library/std/src/lib.rs
--- a/library/std/src/lib.rs
+++ b/library/std/src/lib.rs
@@ -304,6 +304,7 @@
 #![feature(negative_impls)]
 #![feature(optimize_attribute)]
 #![feature(prelude_import)]
+#![feature(rust_cold_cc)]
 #![feature(rustc_attrs)]
 #![feature(rustdoc_internals)]
 #![feature(staged_api)]
diff -ru a/library/std/src/sync/once_lock.rs b/library/std/src/sync/once_lock.rs
--- a/library/std/src/sync/once_lock.rs
+++ b/library/std/src/sync/once_lock.rs
@@ -559,7 +559,7 @@
 
     #[cold]
     #[optimize(size)]
-    fn initialize<F, E>(&self, f: F) -> Result<(), E>
+    extern "rust-cold" fn initialize<F, E>(&self, f: F) -> Result<(), E>
     where
         F: FnOnce() -> Result<T, E>,
     {
